src/videojs-media-sources.js

```diff
@@ -100,64 +98,156 @@
             return videojs.createTimeRange();
           }
-          start = Math.max(this.videoBuffer_.buffered.start(0),
-                           this.audioBuffer_.buffered.start(0));
-          end = Math.min(this.videoBuffer_.buffered.end(0),
-                         this.audioBuffer_.buffered.end(0));
+
+          // TODO: Calculate and return the true intersection of the buffered
+          //       property
+          if (this.videoBuffer_) {
+            lastIndex = this.videoBuffer_.buffered.length - 1;
+            starts.push(this.videoBuffer_.buffered.start(0));
+            ends.push(this.videoBuffer_.buffered.end(lastIndex));
+          }
+          if (this.audioBuffer_) {
+            lastIndex = this.audioBuffer_.buffered.length - 1;
+            starts.push(this.audioBuffer_.buffered.start(0));
+            ends.push(this.audioBuffer_.buffered.end(lastIndex));
+          }
+
+          start = Math.max.apply(Math, starts);
+          end = Math.min.apply(Math, ends);
           return videojs.createTimeRange(start, end);
         }
       });
     },
     appendBuffer: function(segment) {
-      this.audioUpdating_ = this.videoUpdating_ = true;
-      this.transmuxer_.push(segment);
-      this.transmuxer_.flush();
+      // Start the internal "updating" state
+      this.bufferUpdating_ = true;
+
+      this.transmuxer_.postMessage({action: 'push', data: segment.buffer}, [segment.buffer]);
+      this.transmuxer_.postMessage({action: 'flush'});
+    },
+    /**
+     * Process any segments that the muxer has output
+     * Concatenate segments together based on type and append them into
+     * their respective sourceBuffers
+     */
+    processPendingSegments_: function() {
+      var segment, buffer, i, offset,
+        videoSegments = [],
+        audioSegments = [],
+        videoBytes = 0,
+        audioBytes = 0;
+
+      // Sort segments into separate video/audio arrays and
+      // keep track of their total byte lengths
+      while ((segment = this.pendingBuffers_.shift())) {
+        if (segment.type === 'video') {
+          videoSegments.push(segment.data);
+          videoBytes += segment.data.byteLength;
+        } else {
+          audioSegments.push(segment.data);
+          audioBytes += segment.data.byteLength;
+        }
+      }
+
+      // Merge multiple video segments into one and append
+      if (videoBytes) {
+        this.concatAndAppendSegments_(videoSegments, videoBytes, this.videoBuffer_);
+      }
+
+      // Merge multiple audio segments into one and append
+      if (audioBytes) {
+        this.concatAndAppendSegments_(audioSegments,audioBytes, this.audioBuffer_);
+      }
+
+      // We are no longer in the internal "updating" state
+      this.bufferUpdating_ = false;
+    },
+    concatAndAppendSegments_: function(segments, totalBytes, destinationBuffer) {
+      var offset = 0;
+      var tempBuffer = new Uint8Array(totalBytes);
```

Comment:
Multiple `var`s again. I would love if we could just babelify this project so I can stop nagging about it :)